## PREVIEW DATA


In [19]:

USE ChienTD
GO

SELECT TOP 5 
    *
FROM vnbc


Commands completed successfully.

(5 rows affected)

credit_sco | gender | age | occupation           | balance   | monthly_ir | address                | tenure_ye | nums_card | nums_service | active_member | last_active_date | last_transaction_month | created_date | exit | customer_segment | engagement_score | loyalty_level | digital_behavior | risk_score | risk_segment | cluster_group | age_group
-----------+--------+-----+----------------------+-----------+------------+------------------------+-----------+-----------+--------------+---------------+------------------+------------------------+--------------+------+------------------+------------------+---------------+------------------+------------+--------------+---------------+----------
725        | male   | 55  | Chủ Doanh nghiệp nhỏ | 177306004 | 121000000  | Phường An Hội (BT)     | 0         | 4         | 8            | 1             | 2025-04-03       | 91038993               | 2025-02-27   | 0    | Priority         | 90               | Bronze        | mobile           | 359    

# NHÓM 1 : CHÂN DUNG VÀ PHÂN KHÚC

## 1, Tỷ lệ rời bỏ thay đổi như thế nào giữa các nhóm tuổi(age_group) ?

In [4]:
SELECT 
    age_group,
    ROUND(SUM([exit])*100.0/COUNT(*),2) as exit_ratio
FROM 
    vnbc 
GROUP BY age_group
ORDER BY exit_ratio

(4 rows affected)

age_group  | exit_ratio     
-----------+----------------
Boomers    | 15.790000000000
Gen X      | 16.900000000000
Millenials | 20.730000000000
Gen Z      | 26.970000000000
(4 rows)

NHẬN XÉT : 

* Có thể thấy GenZ là nhóm khách hàng có tỷ lệ rời bỏ cao nhất với 27% rồi giảm dần với nhóm tuổi cao dần. Điều này có thể giải thích về tâm lý giữa các nhóm tuổi khi mà nhóm tuổi càng cao thì càng có xu hướng ổn định và gắn bó với 1 ngân hàng nhất định.



## 2,Tỷ lệ rời bỏ giữa các phân khúc(customer_segment)?

In [3]:
SELECT 
    customer_segment,
    ROUND(SUM([exit])*100.0/COUNT(*),2) AS exit_ratio,
    COUNT(*) AS total_customer
FROM 
    vnbc 
GROUP BY customer_segment
ORDER BY exit_ratio

(4 rows affected)

customer_segment | exit_ratio      | total_customer
-----------------+-----------------+---------------
Priority         | 0.070000000000  | 15692         
Affluent         | 3.100000000000  | 11210         
Emerging         | 17.620000000000 | 31662         
Mass             | 39.480000000000 | 21436         
(4 rows)

NHẬN XÉT : 

* Có thể thấy phân khúc đại chúng (Mass) có tỷ lệ rời bỏ khá cao . Điều này khá dễ hiểu khi họ có quá nhiều sự lựa chọn và việc rời bỏ ngân hàng này sang ngân hàng khác rất dễ dàng với nhiều lý do khác nhau...

* Các phân khúc cao cấp hơn như giới thượng lưu và siêu giàu thì tỷ lệ exit khá thấp và gần như là khách hàng trung thành;các ngân hàng thường có các chính sách ưu đãi nhằm giữ chân họ. Số lượng khách hàng siêu giàu với exit thấp đã nói lên việc chăm sóc khách hàng ở phân khúc này rất tốt 

* Khách hàng tiềm năng thường nhạy cảm hơn với các chương trình khuyến mãi từ các đối thủ canh tranh nên tỷ lệ rời bỏ cũng cao hơn (ở mức 17,6 % hay cứ 6 người thì 1 người rời) và đây cũng là nhóm khách hàng chiếm số lượng lớn nhất

==> Các ngân hàng đang có xu hướng giữ chân các khách hàng ở phân khúc cao hơn và chấp nhận exit cao ở phân khúc đại chúng


## 3,Tỷ lệ rời bỏ dựa trên nghề nghiệp

In [2]:
SELECT 
    occupation,
    ROUND(SUM([exit])*100.0/COUNT(*),2) AS exit_ratio,
    COUNT(*) AS total_customer
FROM 
    vnbc 
GROUP BY occupation
ORDER BY exit_ratio

(10 rows affected)

occupation                    | exit_ratio      | total_customer
------------------------------+-----------------+---------------
Chủ Doanh nghiệp nhỏ          | 0.130000000000  | 7915          
Quản lý/Lãnh đạo              | 0.230000000000  | 7986          
Kỹ sư/Chuyên viên IT          | 3.070000000000  | 8199          
Kế toán/Tài chính             | 14.530000000000 | 7972          
Kinh doanh/Bán hàng           | 15.090000000000 | 7967          
Giáo viên/Giảng viên          | 16.430000000000 | 7975          
Nhân viên văn phòng/Công chức | 21.910000000000 | 8061          
Hưu trí                       | 31.470000000000 | 7861          
Lao động phổ thông            | 34.730000000000 | 7968          
Nội trợ/Sinh viên             | 42.530000000000 | 8096          
(10 rows)

NHẬN XÉT : 

* Số lượng khách hàng rất đồng đều ở các vị trí nghề nghiệp khác nhau

* Các vị trí như Chủ Doanh Nghiệp nhỏ hay Quản lý/Lãnh đạo có tỷ lệ exit rất thấp . Điều này là do các Chủ Doanh Nghiệp rất cần có nguồn vốn để phát triển và mở rộng doanh nghiệp của mình và ngân hàng là kênh lý tưởng cũng như sẽ đi cùng với họ trong suốt quá trình đó; mặt khác,đối với quản lý và các lãnh đạo thì họ chọn 1 ngân hàng để sử dụng lâu dài khi mà thủ tục chuyển là vô cùng phiền hà cũng như sự đãi ngộ của ngân hàng với những nguời này là cực tốt.

* Vị trí kỹ sư IT cũng có tỷ lệ exit ở mức thấp(ở mức 3%) do đây là ngành nghề có mức thu nhập cao hơn mặt bằng chung;họ cũng là những người có am hiểu về công nghệ và từ đó họ sẽ chọn ngân hàng tin tưởng đáp ứng được các tiêu chí về công nghệ mà họ đặt ra nhằm gắn bó lâu dài 

* Ngành nghề: Kế toán/Tài chính, Kinh doanh/Bán hàng, Giáo viên/Giảng viên => Đây là nhóm khách hàng ổn định nhất , có tâm lý ưa chuộng sự ổn định và gắn bó lâu dài với 1 ngân hàng nhất định nào đó.

* Nhóm nhân viên văn phòng, công chức : Họ đa phần nhận lương qua tài khoản , họ ở lại vì sự tiện lợi khi nhận lương nhưng sẽ chuyển đi nếu ngân hàng khác có nhiều ưu đãi hơn hoặc phí dịch vụ rẻ hơn

* Nhóm đã về hưu và lao động phổ thông : Những người đã về hưu thường sử dụng ngân hàng để nhận lương hưu hoặc gửi tiết kiệm.Nếu như có ngân hàng khác có lãi suất hấp dẫn hơn họ sẽ rời sang ngay . Mặt khác, những người lao động thường có thu nhập không ổn định hoặc nhận lương tiền mặt,họ không có nhiều ràng buộc về khoản vay hay đầu tư nên thoát ra khá dễ dàng .

* Nhóm nôi trợ,sinh viên : Sinh viên thường rất nhạy về công nghệ,họ sẵn sàng mở 3-4 tài khoản ở nhiều ngân hàng khác nhau chỉ để hưởng quà tằng hoặc trải nghiệm tính năng mới trong khi chỉ thật sự sử dụng 1 ngân hàng.Đối với nội trợ ,họ ưu tiên sự tiện lợi hoặc voucher giảm giá khi mua sắm, không có sự ràng buộc về vay vốn hay đầu tư với ngân hàng

# NHÓM 2 : HÀNH VI VÀ ĐIỂM GẤP KHÚC 

## 1, Có phải càng sử dụng nhiều dịch vụ thì tỷ lệ exit càng giảm không ? Và từ bao nhiêu dịch vụ thì bắt đầu giảm mạnh ?

In [1]:
WITH ServiceAnalysis AS (
    SELECT 
        nums_service,
        COUNT(*) AS total_customers,
        ROUND(SUM([exit]) * 100.0 /COUNT(*),2) AS exit_ratio
    FROM 
        vnbc
    GROUP BY nums_service
)

SELECT 
    nums_service,
    total_customers,
    exit_ratio,
    LAG(exit_ratio) OVER (ORDER BY nums_service) AS prev_exit_ratio,
    ROUND(LAG(exit_ratio) OVER (ORDER BY nums_service) - exit_ratio ,2) AS churn_reduction_impact 
FROM ServiceAnalysis
ORDER BY nums_service;


(8 rows affected)

nums_service | total_customers | exit_ratio      | prev_exit_ratio | churn_reduction_impact
-------------+-----------------+-----------------+-----------------+-----------------------
1            | 13519           | 33.680000000000 | NULL            | NULL                  
2            | 14860           | 25.750000000000 | 33.680000000000 | 7.930000000000        
3            | 15256           | 19.910000000000 | 25.750000000000 | 5.840000000000        
4            | 12800           | 13.400000000000 | 19.910000000000 | 6.510000000000        
5            | 9244            | 8.830000000000  | 13.400000000000 | 4.570000000000        
6            | 6049            | 4.960000000000  | 8.830000000000  | 3.870000000000        
7            | 3743            | 2.590000000000  | 4.960000000000  | 2.370000000000        
8            | 4529            | 1.210000000000  | 2.590000000000  | 1.380000000000        
(8 rows)

NHẬN XÉT : 

* Từ 1 lên 2 dịch vụ là bước làm quen hệ sinh thái và có mức giảm exit_ratio lớn nhất (7.93%)

* Thôg thường , mức giảm exit_ratio sẽ yếu dần theo số lượng dịch vụ .Tuy nhiên, ở mức từ 3 lên 4 dịch vụ thì mức giảm lại tăng từ 5,8% lên 6,5% .Điều này chứng tỏ ở mức 4 là mắt xích quan trọng , khiến cho khách hàng khó rời bỏ khách hàng nếu sử dụng đến mức này . Ngoài ra ,lượng khách hàng sử dụng 4 dịch vụ cũng khá đồng đều so với tệp khách hàng chỉ sử dụng 1 hay 2 dịch vụ 

==> Để thuyết phục khách hàng từ mức 4 lên mức 5 tốn nhiều chi phí về cả marketing và sales hơn nếu so với việc đưa khách hàng từ mức 1,2,3 lên mức 4 .Đây chính là vùng quan trọng nhất không chỉ mang lại lợi nhuận mà còn đem lại sự gắn kết bền vững nhất so với chi phí bỏ ra

## 2, Có phải tất cả các khách hàng đang sử dụng tích cực (active_member = 1) đều có engagement_score cao ? Tỷ lệ rời bỏ sẽ thay đổi như thế nào giữa các nhóm tuổi ?

In [1]:
WITH 
    EngagementRanked AS (
        SELECT 
            age_group,
            active_member,
            [exit],
            engagement_score,
            CUME_DIST() OVER (ORDER BY engagement_score) AS engagement_pct    /* CUME_DIST dùng để tính phân vị */
        FROM vnbc 
    ),
    Classification AS (
        SELECT 
            CASE 
                WHEN active_member = 1 AND engagement_pct <= 0.5 THEN 'virtual_active_customer'
                WHEN active_member = 0 THEN 'truly_inactive'
                ELSE 'healthy_active' 
            END AS customer_classification,
            [exit],
            age_group
        FROM 
            EngagementRanked
    )

SELECT 
    customer_classification,
    age_group,
    COUNT(*) AS total_customer,
    ROUND(AVG(1.0*[exit])*100,2) AS exit_ratio
FROM 
    Classification
WHERE customer_classification IN ('virtual_active_customer','truly_inactive')
GROUP BY customer_classification,age_group



(4 rows affected)

customer_classification | age_group  | total_customer | exit_ratio
------------------------+------------+----------------+-----------
truly_inactive          | Gen Z      | 1357           | 30.950000 
truly_inactive          | Millenials | 18397          | 24.250000 
truly_inactive          | Gen X      | 29511          | 19.860000 
truly_inactive          | Boomers    | 13559          | 18.750000 
(4 rows)

NHẬN XÉT : 

* Như vậy có thể thấy đa số khách hàng vẫn chưa phải là khách hàng tích cực cùng với đó là tỷ lệ exit khá đáng báo động

* Ở nhóm tuổi GenX chiếm số lượng khách hàng rất lớn nhưng tỷ lệ exit nếu inactive vẫn rất cao lên đến 20% trong khi vốn dĩ họ là nhóm khách hàng 'ổn định' và có tài chính vững nhất trong các nhóm tuổi .Điều này có thể do chuyển sang sử dụng dịch vụ các đối thủ cạnh tranh và điều này gây thất thoát lợi nhuận rất lớn

==> Cần có những chiến dịch 'Re-activate' đối với những khách hàng inactive - đặc biệt với GenX để biến họ thành những khách hàng tích cực vì chi phí bỏ ra sẽ thấp hơn nhiều so với tìm kiếm khách hàng mới .

## 3, Liệu độ trung thành(loyalty_level) cao và cách tương tác của khách hàng là online có làm tăng engagement_score hay không? Tỷ lệ exit như nào ?

In [20]:
SELECT 
    loyalty_level,
    digital_behavior,
    AVG(engagement_score) AS avg_engagement_scr,
    COUNT(*) AS total_customer,
    ROUND(AVG(1.0*[exit])*100,2) AS exit_ratio
FROM vnbc
GROUP BY loyalty_level,digital_behavior
ORDER BY avg_engagement_scr,loyalty_level


(5 rows affected)

loyalty_level | digital_behavior | avg_engagement_scr | total_customer | exit_ratio
--------------+------------------+--------------------+----------------+-----------
Bronze        | offline          | 19                 | 58726          | 22.510000 
Silver        | offline          | 38                 | 4098           | 1.610000  
Bronze        | mobile           | 73                 | 10526          | 7.630000  
Gold          | mobile           | 73                 | 1771           | 2.820000  
Silver        | mobile           | 73                 | 4879           | 5.410000  
(5 rows)

NHẬN XÉT : 

* Có thể thấy được số lượng người dùng trung thành ở mức thấp nhất chiếm đa số với điểm trung bình tham gia rất thấp với chỉ con số 19 dẫn đến tỷ lệ rời bỏ cũng rất cao ở mức 22,5 %. Điều này còn có thể chứng minh thông qua việc đây là nhóm khách hàng truyền thống và thường xuyên giao dịch trực tiếp tại quầy nên rất dễ đổi sang các ngân hàng khác có chính sách tốt hơn cũng như không thể dùng các dịch vụ số tiện lợi mà ngân hàng phát triển để níu chân họ.

* Mặt khác ,với các khách hàng có mức độ trung thành là 'Bạc' tuy số lượng chiếm không quá lớn nhưng tỷ lệ exit lại giảm đến đáng kinh ngạc chỉ ở mức 1,61% .Điều này chứng tỏ họ là tệp khách hàng bắt đầu quen với việc sử dụng dịch vụ tại quầy và xem ngân hàng đang dùng là liên kết lâu dài.Thông thường ,tệp khách hàng này thường là các chủ doanh nghiệp nhỏ hoặc quản lý hay trao niềm tin với lượng vốn lớn vào 1 ngân hàng cố định 

* Khi bước sang giao dịch số ,có thể thấy được điểm tham gia trung bình của cả 3 tệp khách hàng đồng đều với nhau(73) và tỷ lệ rời bỏ giảm xuống mức vô cùng ấn tượng .Với việc được tiếp xúc những công nghệ và tiện ích khiến cho việc giao dịch và thủ tục trở nên tiện lợi hơn thì họ sẽ có xu hướng gắn bó với ngân hàng đó luôn và giảm dần dựa trên mức độ tăng sự trung thành .

* Với việc chuyển đổi số,các giao dịch trở nên nhanh và tiện lợi hơn rất nhiều ,điều này rõ ràng khuyến khích họ tham gia  giao dịch với tần suất lớn hơn và thường xuyên hơn

==> Có thể thấy một vài năm trở lại đây thì các ngân hàng đang tích cực chuyển đổi số và các con số trên là minh chứng cho sự hiệu quả của chiến dịch này .Mặt khác, đa số người dân vẫn đang chọn cách giao dịch truyền thống vì nhiều lý do khác nhau như sự tin tưởng hay bảo mật dữ liệu ,... Ở phân khúc khách hàng này thì các ngân hàng có thể ưu tiên nâng độ trung thành của họ lên mức 'Bạc' với hiệu quả đã được chứng minh ở trên

# NHÓM 3 : GIÁ TRỊ BỊ MẤT MÁT

## 1, Tổng tài sản mất mát (AUM)

In [1]:
SELECT 
    customer_segment,
    COUNT(*) AS total_customer_exit,
    FORMAT(SUM(balance), 'N0') AS total_AUN_lost,
    FORMAT(AVG(balance), 'N0') AS avg_AUN_lost
FROM vnbc  
WHERE [exit] = 1
GROUP BY customer_segment
ORDER BY total_AUN_lost DESC

(4 rows affected)

customer_segment | total_customer_exit | total_AUN_lost  | avg_AUN_lost
-----------------+---------------------+-----------------+-------------
Affluent         | 347                 | 20,259,120,433  | 58,383,632  
Emerging         | 5580                | 157,780,710,700 | 28,276,113  
Mass             | 8462                | 103,132,565,193 | 12,187,729  
Priority         | 11                  | 1,203,790,549   | 109,435,504 
(4 rows)

NHẬN XÉT : 

* Mặc dù về số lượng khách hàng thì phân khúc đại chúng vẫn tỷ trọng lớn nhất nhưng phân khúc khách hàng tiềm năng lại gây thất thoát của ngân hàng gấp 1,5 lần về cả tổng lẫn trung bình (157,8 tỷ so với 103,1 tỷ) .Bên cạnh đó ,mỗi khách hàng phân khúc này gây thất thoát của ngân hàng đến 28 triệu nếu rời bỏ.

* Giới thượng lưu và siêu giàu (Affluent && Priority) có số lượng ít hơn đáng kể nhưng chỉ cần 1 người rời cũng có thể khiến cho tài sản mất mát rất nhanh .Với phân khúc Priority thì tuy chỉ có 11 người nhưng mỗi người rời bỏ khiến cho ngân hàng mất đến 109,5 triệu - 1 con số rất lớn.

==> Ngân hàng nên có những chính sách ưu đãi đặc biệt nhằm dữ chân những khách hàng tiềm năng bởi họ là nhóm có khả năng rời đi chỉ sau nhóm đại chúng tuy nhiên lại là nhóm gây thất thoát nhiều nhất .Còn đối với phân khúc giàu có,cần có thêm những chương trình đặc biệt (Về lãi suất, đặc quyền,...) để kéo họ ở lại với ngân hàng

## 2, Tỷ lệ rời bỏ giữa thu nhập hàng tháng - nghề nghiệp 

In [21]:
WITH 
    binned_income AS(   
        SELECT 
            occupation,
            [exit],
            CASE 
                WHEN monthly_ir < 15000000 THEN 'LOW'
                WHEN monthly_ir BETWEEN 15000000 AND 45000000 THEN 'MIDDLE'
                WHEN monthly_ir BETWEEN 45000000 AND 100000000 THEN 'HIGH'
                ELSE 'ELITE' 
            END AS income_segment
        FROM 
            vnbc
    )

SELECT 
    occupation AS [occupation\exit_ratio],
    ROUND(AVG(CASE WHEN income_segment = 'LOW' THEN CAST([exit] AS FLOAT) * 100.0 END),2) AS 'Low_Income_Exit_%',
    ROUND(AVG(CASE WHEN income_segment = 'MIDDLE' THEN CAST([exit] AS FLOAT) * 100.0 END),2) AS 'Middle_Income_Exit_%',
    ROUND(AVG(CASE WHEN income_segment = 'HIGH' THEN CAST([exit] AS FLOAT) * 100.0 END),2) AS 'High_Income_Exit_%',
    ROUND(AVG(CASE WHEN income_segment = 'ELITE' THEN CAST([exit] AS FLOAT) * 100.0 END),2) AS 'Elite_Income_Exit_%'
FROM 
    binned_income
GROUP BY occupation
ORDER BY occupation

(10 rows affected)

occupation\exit_ratio         | Low_Income_Exit_% | Middle_Income_Exit_% | High_Income_Exit_% | Elite_Income_Exit_%
------------------------------+-------------------+----------------------+--------------------+--------------------
Chủ Doanh nghiệp nhỏ          | NULL              | 50                   | 0,24               | 0                  
Giáo viên/Giảng viên          | 30,53             | 15,34                | NULL               | NULL               
Hưu trí                       | 39,09             | 22,23                | NULL               | NULL               
Kế toán/Tài chính             | 29,19             | 13,68                | 2,78               | NULL               
Kinh doanh/Bán hàng           | 34,15             | 12,07                | 2,02               | NULL               
Kỹ sư/Chuyên viên IT          | NULL              | 5,81                 | 0,71               | NULL               
Lao động phổ thông            | 37,23             | 27,1                

NHẬN XÉT : 

* Đối với nhóm khách hàng Chủ Doanh Nghiệp và Quản lý/Lãnh đạo ,thu nhập của họ luôn từ Middle trở lên khi mà tỷ lệ Low_Income_Exit là giá trị NULL. Họ là phân khúc có thu nhập lớn hơn phần còn lại và một khi đang sử dụng 1 ngân hàng cố định với mức lương rất cao (Elite) thì họ chắc chắn không rời bỏ.Mặt khác thì Chủ Doanh nghiệp nhỏ ở mức thu nhập ở mức Middle thì có tỷ lệ rời lên đến 50% - điều này có thể giải thích với các lý do như rủi ro không tương xứng cũng như khát vốn nhưng lại khó tiếp cận  . Đối với các quản lý / lãnh đạo, họ có tỷ lệ exit chỉ là 2,4% nếu ở mức thu nhập Middle do họ không phải chịu rủi ro nếu như công ty làm ăn thua lỗ .Bên cạnh đó, thông thường các tài khoản ngân hàng là do cty chỉ định cũng như đặc thù công việc bận rộn nên họ không có nhiều thời gian suy nghĩ đổi ngân hàng

* Các kỹ sư IT có thu nhập hàng tháng cao hơn mặt bằng chung khi mà không có ai lương ở mức Low và tỷ lệ rời bỏ của họ cũng thấp ở mức 5,81%. Sở dĩ có điều này là do sự chọn lọc kỹ càng ngay từ ban đầu với những yêu cầu về tốc độ ,bảo mật, tính tự động ,... Họ cũng là phân khúc sẽ được hưởng những ưu đãi tốt từ ngân hàng.

* Các nhóm khách hàng như Giáo viên/Giảng viên , Kinh doanh/Bán hàng thì có tỷ lệ rời bỏ giảm dần theo thu nhập (Từ Low -> High) .Ở mức thu nhập thấp ,họ sẵn sàng rời bỏ để chuyển sang ngân hàng có chính sách ưu đãi hay các chương trình khuyến mãi tốt hơn. Mặt khác, khi tăng mức thu nhập lên Medium và High thì họ xu hướng dùng nhiều dịch vụ của ngân hàng hơn như gửi tiết kiệm hay vay tiêu dùng và gắn bó vơi 1 ngân hàng nhất định mà họ tin tưởng.

* Lao động phổ thông ,nội trợ và sinh viên, hưu trí có tỷ lệ rời bỏ cao nhất trong các ngành nghề và đặc thù của nhóm này là thu nhập hàng tháng chỉ dao động trong khoảng Low và Middle. Họ không có sự ràng buộc nào với ngân hàng : không khoản vay , không có thẻ tín dụng ,không có bảo hiểm qua ngân hàng,... Họ dễ dàng chuyển qua ngân hàng khác khiến là ngân hàng đó đáp ứng được 1 tiêu chí nào đó mà họ đưa ra.